src/Model/Order/Order.php

```diff
@@ -131,11 +132,26 @@ public static function create(
 
     private function applyOrderWasCreated(OrderWasCreated $event): void
     {
-        $this->id              = $event->getId();
-        $this->businessName    = $event->getBusinessName();
+        $this->id = $event->getId();
+        $this->businessName = $event->getBusinessName();
         $this->businessAddress = $event->getBusinessAddress();
-        $this->contactPerson   = $event->getContactPerson();
-        $this->isPaid          = $event->getHasPaid();
-        $this->amount          = $event->getAmount();
+        $this->contactPerson = $event->getContactPerson();
+        $this->isPaid = $event->getIsPaid();
+        $this->amount = $event->getAmount();
+    }
+
+    public function pay(): void
+    {
+        $this->recordThat(
+            OrderWasPaid::occur($this->getId()->toNative(), [
+                'isPaid' => NonNullOrderIsPaid::true()
+            ])
+        );
+    }
+
+    private function applyOrderWasPaid(OrderWasPaid $event): void
+    {
+        $this->id = $event->getId();
+        $this->isPaid = NonNullOrderIsPaid::true();
```

Comment:
Following on from the comment in `OrderTest`, you would add something like `$this->datePaid = NonNullOrderDatePaid::fromNative($event->createdAt())`

Reply:
The cool thing about event sourcing is that if this had gone live and orders paid without `datePaid` field then we wouldn't have actually lost any information.

We could just add the `datePaid` field and then replay all the events. The `createdAt` is in all events so it would automatically add it in to all the existing orders.